NullPointerException in ExpressionHandler

When parsing the vsomeip stack (https://github.com/GENIVI/vsomeip), I get the the following java NullPointerException:

```
08:59:48.020 [ForkJoinPool.commonPool-worker-3] ERROR de.fraunhofer.aisec.cpg.frontends.Handler - file:///home/tobias/Downloads/vsomeip/implementation/endpoints/src/server_endpoint_impl.cpp:89:29: Parsing of type class org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTLambdaExpression is not supported (yet)
08:59:48.027 [ForkJoinPool.commonPool-worker-3] ERROR de.fraunhofer.aisec.cpg.passes.scopes.ScopeManager - Cannot get current Record. No scope.
java.util.concurrent.ExecutionException: java.lang.NullPointerException
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
        at de.fraunhofer.aisec.cpg.ptn4j.Application.call(Application.kt:243)
        at de.fraunhofer.aisec.cpg.ptn4j.Application.call(Application.kt:30)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1783)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2141)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2108)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1975)
        at picocli.CommandLine.execute(CommandLine.java:1904)
        at de.fraunhofer.aisec.cpg.ptn4j.ApplicationKt.main(Application.kt:253)
Caused by: java.lang.NullPointerException
        at de.fraunhofer.aisec.cpg.frontends.cpp.ExpressionHandler.handleFieldReference(ExpressionHandler.java:300)
        at de.fraunhofer.aisec.cpg.frontends.cpp.ExpressionHandler.lambda$new$5(ExpressionHandler.java:73)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.ExpressionHandler.handleFunctionCallExpression(ExpressionHandler.java:394)
        at de.fraunhofer.aisec.cpg.frontends.cpp.ExpressionHandler.lambda$new$6(ExpressionHandler.java:76)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.InitializerHandler.handleEqualsInitializer(InitializerHandler.java:77)
        at de.fraunhofer.aisec.cpg.frontends.cpp.InitializerHandler.lambda$new$1(InitializerHandler.java:49)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclaratorHandler.handleDeclarator(DeclaratorHandler.java:97)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclaratorHandler.lambda$new$0(DeclaratorHandler.java:65)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.handleSimpleDeclaration(DeclarationHandler.java:308)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.lambda$new$1(DeclarationHandler.java:89)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleDeclarationStatement(StatementHandler.java:310)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$2(StatementHandler.java:97)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleCompoundStatement(StatementHandler.java:338)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$0(StatementHandler.java:93)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleIfStatement(StatementHandler.java:179)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$4(StatementHandler.java:101)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleCompoundStatement(StatementHandler.java:338)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$0(StatementHandler.java:93)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleIfStatement(StatementHandler.java:179)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$4(StatementHandler.java:101)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleCompoundStatement(StatementHandler.java:338)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$0(StatementHandler.java:93)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleIfStatement(StatementHandler.java:179)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$4(StatementHandler.java:101)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.handleCompoundStatement(StatementHandler.java:338)
        at de.fraunhofer.aisec.cpg.frontends.cpp.StatementHandler.lambda$new$0(StatementHandler.java:93)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.handleFunctionDefinition(DeclarationHandler.java:204)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.lambda$new$2(DeclarationHandler.java:92)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.handleTemplateDeclaration(DeclarationHandler.java:260)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.lambda$new$0(DeclarationHandler.java:86)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.handleNamespace(DeclarationHandler.java:116)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.lambda$new$4(DeclarationHandler.java:98)
        at de.fraunhofer.aisec.cpg.frontends.Handler.handle(Handler.java:111)
        at de.fraunhofer.aisec.cpg.frontends.cpp.DeclarationHandler.handleTranslationUnit(DeclarationHandler.java:369)
        at de.fraunhofer.aisec.cpg.frontends.cpp.CXXLanguageFrontend.parse(CXXLanguageFrontend.java:291)
        at de.fraunhofer.aisec.cpg.TranslationManager.runFrontends(TranslationManager.java:241)
        at de.fraunhofer.aisec.cpg.TranslationManager.lambda$analyze$0(TranslationManager.java:98)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```
Maybe the error is related to the `christian/typeparser-rework` branch.
At least, when using this branch, the error does not appear, or is handled.
As far as I can see, the changes have been merged to cpg v3, but still, there seems to be a problem.

Can you reproduce the bug?

--- a/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/ExpressionHandler.java
+++ b/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/ExpressionHandler.java
@@ -292,12 +292,15 @@ class ExpressionHandler extends Handler<Expression, IASTInitializerClause, CXXLa
   private Expression handleFieldReference(CPPASTFieldReference ctx) {
     Expression base = this.handle(ctx.getFieldOwner());
     // Replace Literal this with a reference pointing to this
-    if (base instanceof Literal && ((Literal) base).getValue().equals("this")) {
+    if (base instanceof Literal && ((Literal<?>) base).getValue().equals("this")) {
       PhysicalLocation location = base.getLocation();
+
+      var record = lang.getScopeManager().getCurrentRecord();
+
       base =
           NodeBuilder.newDeclaredReferenceExpression(
               "this",
-              lang.getScopeManager().getCurrentRecord().getThis().getType(),
+              record != null ? record.getThis().getType() : UnknownType.getUnknownType(),
               base.getCode());
       base.setLocation(location);
     }

